test/src/index/transport/testBrowserstackTransport.js

```diff
@@ -341,7 +341,80 @@ describe('BrowserstackTransport', function () {
     assert.strictEqual(transport.buildId, '123-567-89');
 
   });
+  
+  it('test create Transport for Browserstack - Automate TurboScale', async function() {
+    const client = NightwatchClient.client({
+      webdriver: {
+        host: 'hub-cloud.browserstack-ats.com',
+        port: 443,
+        start_process: true
+      },
+      desiredCapabilities: {
+        'browserstack.user': 'test-access-user',
+        'browserstack.key': 'test-access-key',
+        browserName: 'chrome'
+      }
+    });
 
+    nock('https://hub-cloud.browserstack-ats.com')
+      .post('/wd/hub/session')
+      .reply(201, function (uri, requestBody) {
+        return {
+          value: {
+            sessionId: '1352110219202',
+            capabilities: requestBody.capabilities
+          }
+        };
+      });
+    
+    nock('https://api.browserstack.com')
+      .get('/automate-turboscale/v1/builds?status=running')
+      .reply(200, {
+        'builds': [
+          {
+            name: 'nightwatch-test-build',
+            hashed_id: '123-567-89'
+          },
+          {
+            name: 'test-build'
+          }
+        ]
+      });
+    assert.ok(client.transport instanceof AutomateTurboScale);
+    assert.strictEqual(client.settings.webdriver.host, 'hub-cloud.browserstack-ats.com');
+    assert.strictEqual(client.settings.webdriver.default_path_prefix, '/wd/hub');
+    assert.strictEqual(client.settings.webdriver.ssl, true);
+
+    const {transport} = client;
+    assert.ok(transport instanceof SeleniumRemote);
+
+    let result = await transport.createSession({argv: undefined, moduleKey: ''});
+    result.sessionId = '1234567';
```

Comment:
why overwrite the `sessionId` here?

Reply:
Instead assert that the sessionId here should be equal to that used in the first `nock` call above: `1352110219202` and leave it as it is.